pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/fwd/NoOpForwardIndexCreator.java

```diff
@@ -19,12 +19,13 @@
 package org.apache.pinot.segment.local.segment.creator.impl.fwd;
 
 import java.io.IOException;
+import java.math.BigDecimal;
 import org.apache.pinot.segment.spi.index.creator.ForwardIndexCreator;
 import org.apache.pinot.spi.data.FieldSpec;
 
 
 /**
- * Forward index creator for dictionary-encoded single and multi-value columns with forward index disabled.
+ * Forward index creator for dictionary-encoded and raw single and multi-value columns with forward index disabled.
  * This is a no-op.
  */
 public class NoOpForwardIndexCreator implements ForwardIndexCreator {
```

Comment:
Suggest getting rid of this class and handle disabled forward index explicitly instead of still dumping the values into the no-op creator for both readability and performance concern

Reply:
Sure Jackie, but let me handle removing this as a separate change rather than mixing this into this one.

Reply:
@Jackie-Jiang I've removed this as part of this PR itself